corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/views.js

```diff
@@ -243,6 +245,39 @@ hqDefine("cloudcare/js/formplayer/menus/views", function () {
             };
         },
 
+        iconClick: function (e) {
+            e.stopImmediatePropagation();
+            const origin = window.location.origin;
+            const currentUrlToObject = formplayerUtils.currentUrlToObject();
+            const user = FormplayerFrontend.getChannel().request('currentUser');
+            const appId = currentUrlToObject.copyOf;
+            const domain = user.domain;
+            const caseId = this.model.get('id');
+            const urlTemplate = this.options.endpointActions[1]['urlTemplate'];
+            const actionUrl = origin + urlTemplate
+                .replace("{domain}", domain)
+                .replace("{appid}", appId)
+                .replace("{case_id}", caseId);
```

Comment:
This is probably OK for now but in future we may want to use something like https://www.npmjs.com/package/url-template

Reply:
Thanks, I didn't love this approach. I'll circle back and update it with a followup.